cmd/eksctl-anywhere/cmd/createcluster.go

```diff
@@ -206,10 +207,8 @@ func (cc *createClusterOptions) directoriesToMount(clusterSpec *cluster.Spec, cl
 	if fluxConfig == nil || fluxConfig.Spec.Git == nil {
 		return dirs
 	}
-
-	if cliConfig.GitPrivateKeyFile != "" {
-		dirs = append(dirs, cliConfig.GitPrivateKeyFile)
-	}
+	dirs = append(dirs, filepath.Dir(cliConfig.GitPrivateKeyFile))
+	dirs = append(dirs, filepath.Dir(cliConfig.GitKnownHostsFile))
```

Comment:
we now need to mount a known hosts file into the cli tools container in order to securely bootstrap flux; the user will provide an env var, `EKSA_GIT_KNOWN_HOSTS`, which points to a known hosts file; we then mount that file into the tools container and tell the flux execution to look forthe known hosts at that file path -- for some background, https://security.stackexchange.com/a/20710 is useful.

Reply:
why remove the git private key if condition? we are mounting an empty string?

Reply:
we now have a validation which confirms that the private key file is non-empty at a higher level, since it's required, so we don't need to check that here